src/main/java/appeng/spatial/SpatialDimensionManager.java

```diff
@@ -18,182 +18,98 @@
+            .func_240903_a_(Registry.DIMENSION_KEY, AppEng.makeId("storage_cell"));
 
-    private static final String DIM_ID_PREFIX = "spatial_";
+    public static final RegistryKey<World> WORLD_ID = RegistryKey
+            .func_240903_a_(Registry.WORLD_KEY, AppEng.makeId("storage_cell"));
 
-    private SpatialDimensionManager() {
-    }
+    public static final SpatialDimensionManager INSTANCE = new SpatialDimensionManager();
 
-    @Override
-    public ServerWorld getWorld(RegistryKey<World> cellDim) {
-        MinecraftServer server = getServer();
-        return server.getWorld(cellDim);
+    private SpatialDimensionManager() {
     }
 
-    @Override
-    public RegistryKey<World> createNewCellDimension(BlockPos size) {
-        ResourceLocation dimKey = findFreeDimensionId();
-        AELog.info("Allocating storage cell dimension '%s'", dimKey);
-
-        DynamicDimensions dynamicDimensions = (DynamicDimensions) getServer();
-
-        RegistryKey<World> worldId = RegistryKey.func_240903_a_(Registry.WORLD_KEY, dimKey);
-
-        ServerWorld world = dynamicDimensions.addWorld(worldId, STORAGE_DIMENSION_TYPE, StorageChunkGenerator.INSTANCE);
-
-        SpatialDimensionExtraData spatialExtraData = world.getSavedData().getOrCreate(SpatialDimensionExtraData::new,
-                SpatialDimensionExtraData.ID);
-        spatialExtraData.setSize(size);
-
-        return worldId;
+    public static List<Map.Entry<RegistryKey<Dimension>, Dimension>> removeDimension(List<Map.Entry<RegistryKey<Dimension>, Dimension>> dimensions) {
+        dimensions.removeIf(e -> e.getKey().equals(STORAGE_DIMENSION));
+        return dimensions;
     }
 
     /**
-     * Tries finding the next free storage cell dimension ID based on the currently
-     * registered storage cell dimensions.
+     * Gets the world used to store spatial storage cell's content.
      */
-    private ResourceLocation findFreeDimensionId() {
-        int maxId = 0;
-        for (RegistryKey<World> worldId : getServer().func_240770_D_()) {
-            ResourceLocation regName = worldId.func_240901_a_();
-            if (regName == null || !AppEng.MOD_ID.equals(regName.getNamespace())) {
-                continue;
-            }
-
-            String path = regName.getPath();
-            if (!path.startsWith(DIM_ID_PREFIX)) {
-                continue;
-            }
-
-            try {
-                String numericIdPart = path.substring(DIM_ID_PREFIX.length());
-                maxId = Math.max(Integer.parseUnsignedInt(numericIdPart), maxId);
-            } catch (NumberFormatException e) {
-                AELog.warn("Unparsable storage cell dimension id '%s'", path, e);
-            }
+    public ServerWorld getWorld() {
+        MinecraftServer server = getServer();
+        ServerWorld world = server.getWorld(WORLD_ID);
+        if (world == null) {
+            throw new IllegalStateException("The storage cell world is missing.");
```

Comment:
Probably shorten these with `PreConditions.checkState(world == null, "The storage cell world is missing.")`.
A static import here might even be an idea.